### PR TITLE
fix(inputs.phpfpm): Add timeout for fcgi

### LINKED ISSUE #15035: PHP-FPM input never timeout with FCGI protocol
### Relevant telegraf.conf

```toml
[[inputs.phpfpm]]
  urls = ["fcgi://127.0.0.1:9008/status"]
  timeout = "5s"
```

### Logs from Telegraf

```text
$ ./usr/bin/telegraf --test --config ./phpfpm.conf 
2024-03-21T10:22:07Z I! Loading config: ../phpfpm.conf
2024-03-21T10:22:07Z I! Starting Telegraf 1.30.0 brought to you by InfluxData the makers of InfluxDB
2024-03-21T10:22:07Z I! Available plugins: 233 inputs, 9 aggregators, 31 processors, 24 parsers, 60 outputs, 5 secret-stores
2024-03-21T10:22:07Z I! Loaded inputs: phpfpm
2024-03-21T10:22:07Z I! Loaded aggregators: 
2024-03-21T10:22:07Z I! Loaded processors: 
2024-03-21T10:22:07Z I! Loaded secretstores: 
2024-03-21T10:22:07Z W! Outputs are not used in testing mode!
2024-03-21T10:22:07Z I! Tags enabled: host=MBP-de-pierre
[ Hang ever, forever ]
```

### System info

Telegraf 1.30, MacOS 14.4

### Docker

docker run -p 127.0.0.1:9008:9008 --rm -ti busybox nc -l -p 9008 -v 

### Steps to reproduce

1. Run a FGCI server that never reply. The easiest is to run `nc -l` and let it ignore request
2. Run Telegraf --test
3. Telegraf never complet. You can't ctrl+c it.

The above as commands:

```
$ docker rm -f fcgi-that-never-reply
$ docker run -p 127.0.0.1:9008:9008 -d --name fcgi-that-never-reply busybox sh -c "mkfifo /tmp/pipe; nc -lk -p 9008 -v -e cat /tmp/pipe"
$ cat > phpfpm.conf << EOF
[[inputs.phpfpm]]
  urls = ["fcgi://127.0.0.1:9008/status"]
  timeout = "5s"
EOF
$ ./usr/bin/telegraf --test --config ./phpfpm.conf 
```

### Expected behavior

Terminate after the configured timeout with no result.

### Actual behavior

Never terminate and can't stop it with ctrl+c.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by PierreF**
## Summary

The input PHP-FPM don't have timeout when talking (F)CGI protocol, which could lead to input blocked forever.

The timeout exists and was used for HTTP, but not for fcgi.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #15035

### CODE REVIEW

**Review by srebhan — COMMENTED** [review]
@PierreF thanks for fixing this issue! Could you please add a unit-test for the timeout?!?

**Comment by PierreF** [comment]
I've added the test in the same commit (I'm unsure if you prefer a single commit with code & test or if you prefer avoid push force).
The error seems unrelated to my change: it's on syslog input and phpfpm input test passed.

File: plugins/inputs/phpfpm/phpfpm_test.go [hunk 1]
```diff
@@ -150,6 +151,47 @@ func TestPhpFpmGeneratesMetrics_From_Fcgi(t *testing.T) {
...
+		Urls:    []string{"fcgi://" + tcp.Addr().String() + "/status"},
+		Timeout: config.Duration(timeout),
+		Log:     &testutil.Logger{},
+	}
+	require.NoError(t, r.Init())
+
+	start := time.Now()
+
+	var acc testutil.Accumulator
+	require.NoError(t, acc.GatherError(r.Gather))
```
[the remark this answers is not in the document]
Comment (PierreF) [reply plugins/inputs/phpfpm/phpfpm_test.go]: This is fixed. I've done the same thing that was previously done with only some error being returned as error.

**Review by srebhan — APPROVED** [review]
Awesome! Thanks @PierreF!

**Review by powersj — APPROVED** [review]

**Pull request merged**